flutter/lib/ui/painting.dart

```diff
@@ -239,34 +238,69 @@ enum PaintingStyle {
 /// Most APIs on [Canvas] take a [Paint] object to describe the style
 /// to use for that operation.
 class Paint {
-  /// Whether to paint inside shapes, the edges of shapes, or both.
-  ///
-  /// If null, defaults to [PaintingStyle.fill].
-  PaintingStyle style;
-  static const PaintingStyle _kDefaultStyle = PaintingStyle.fill;
+  Paint() {
```

Comment:
Add a big comment at the top of this class that explains the approach here, so that readers can get oriented before jumping in to bit-twiddling code.